lib/modules/platform/github/index.ts

```diff
@@ -166,14 +171,31 @@ export async function initPlatform({
     }
   }
   logger.debug({ platformConfig, renovateUsername }, 'Platform config');
-  const platformResult: PlatformResult = {
+  const githubDS: HostRule[] = [];
+  const gitHubRule: HostRule[] = [
+    {
+      matchHost: '.pkg.github.com',
+    },
+    {
+      matchHost: 'ghcr.io',
+    },
+  ];
+  for (const rule of gitHubRule) {
+    (['token'] as 'token'[]).forEach((field) => {
+      if (token) {
+        rule[field] = token;
+      }
+    });
+    githubDS.push(rule);
+    hostRules.add(rule);
+  }
+  return {
```

Comment:
should only be done when endpoint is the GitHub.com, otherwise we would try to use a ghes token for wrong platform.

i also think we need to use basic auth for ghcr.io authentication. have you validated the hostrule?

Reply:
i agree with the Endpoint = github.com

as far as i can see in this example: https://github.com/plhw/hf-api-client/blob/54981ffc62467eec0577dd8ff26613c5d18ada7b/renovate.json

the host rule is used the same way for github and ghcr.io, what do you mean by basic auth?

Reply:
those are partially wrong. you can't use username with token.

also the dockerfile host type doesn't exist.

maybe ghcr also supports bearer auth, so it works with token.

but if you check the GitHub docs, they told to use account name as username and token as password for docker cli.

that's why I thought we need to set username and password hostrule. 🤷‍♂️

Reply:
i will close this PR and open a new from my account, maroon is no longer maintaining this.